drivers/interrupt_controller/Kconfig.clic

```diff
@@ -1,18 +1,11 @@
-# Nuclei ECLIC interrupt-controller configuration
-
 # Copyright (c) 2021 Tokita, Hiroshi <tokita.hiroshi@gmail.com>
 # SPDX-License-Identifier: Apache-2.0
 
 DT_COMPAT_NUCLEI_ECLIC = nuclei,eclic
 
-config HAS_NUCLEI_ECLIC
-	bool
-	help
-	  Indicate that the platform has ECLIC.
-
 config NUCLEI_ECLIC
 	bool "Enhanced Core Local Interrupt Controller (ECLIC)"
 	default $(dt_compat_enabled,$(DT_COMPAT_NUCLEI_ECLIC))
-	depends on HAS_NUCLEI_ECLIC
+	depends on RISCV_HAS_CLIC
```

Comment:
`depends on RISCV_HAS_CLIC && SOC_GD32VF103`

Reply:
why? It's up to the `GD32VF103` having a `RISCV_HAS_CLIC` and selecting the correct driver for that.

Reply:
because other SoCs may select `RISCV_HAS_CLIC`, but NUCLEI_ECLIC is now specific to GD32VF103, right?

Reply:
`RISCV_HAS_CLIC` is telling that the platform is shipping some kind of interrupt controller that is based on the CLIC specs, so you can have multiple SoCs selecting that.

`NUCLEI_ECLIC` is a special implementation of the CLIC that is shipped by GD32VF103 but we cannot exclude that it can be shipped by other SoCs as well.

So the rationale here is: the SoC is setting `RISCV_HAS_CLIC` and the specific implementation of CLIC is selected either by defconfig or by the `dt_compat_enabled` macro.

Reply:
but if 2 SoC select `RISCV_HAS_CLIC`, NUCLEI_ECLIC will be selectable even if the SoC is not based on Nuclei. GD32VF103 isn't great, but afaik we don't have Nuclei specific options.

Reply:
> > but if 2 SoC select RISCV_HAS_CLIC, NUCLEI_ECLIC will be selectable even if the SoC is not based on Nuclei.
> 
> Right but if selected manually the user will be facing a compile-time error. The driver selection should be done at Kconfig or DT level IMO otherwise for each new SoC added we should go through all the drivers and add a `depends on $new_soc` for all the supported drivers.

The problem here is that we have no non-SoC Kconfig option to filter this option. You'll get a compile error, sure, but we can fail earlier.

> 
> > GD32VF103 isn't great, but afaik we don't have Nuclei specific options.
> 
> Not sure I got what you mean here.

GD32VF103 is the only SoC with Nuclei IP, so this is probably nobody bothered about adding Nuclei IP options, e.g. `HAS_NUCLEI_ECLIC`, which could be used to make the driver available/unavailable (instead of using SoC specific filter)

Reply:
> The problem here is that we have no non-SoC Kconfig option to filter this option. You'll get a compile error, sure, but we can fail earlier.

But why we should always have a SoC Kconfig option to filter out drivers? Take for example `Kconfig.gic` we are filtering on `CPU_CORTEX` (and indeed we could filter on `RISCV` here) but we are not filtering on SoCs anywhere.

> GD32VF103 is the only SoC with Nuclei IP, so this is probably nobody bothered about adding Nuclei IP options, e.g. HAS_NUCLEI_ECLIC, which could be used to make the driver available/unavailable (instead of using SoC specific filter)

I reeeeally would like to avoid adding more symbols around very SoC specific used only to filter out drivers :)

I propose to summon @stephanosio to help us out here :magic_wand:

Reply:
> > The problem here is that we have no non-SoC Kconfig option to filter this option. You'll get a compile error, sure, but we can fail earlier.
> 
> But why we should always have a SoC Kconfig option to filter out drivers? Take for example `Kconfig.gic` we are filtering on `CPU_CORTEX` (and indeed we could filter on `RISCV` here) but we are not filtering on SoCs anywhere.

Many intc drivers `depend on SOC_`, see e.g. stm32, sam0, gd32, etc. Problem with Nuclei is that is not an SoC... It's an IP provider.

> 
> > GD32VF103 is the only SoC with Nuclei IP, so this is probably nobody bothered about adding Nuclei IP options, e.g. HAS_NUCLEI_ECLIC, which could be used to make the driver available/unavailable (instead of using SoC specific filter)
> 
> I reeeeally would like to avoid adding more symbols around very SoC specific used only to filter out drivers :)
> 
> I propose to summon @stephanosio to help us out here magic_wand

Sounds good, anyway, not gonna block on that, just wanted to see if we can do better than a compile error.